superset/stats_logger.py

```diff
@@ -51,24 +63,30 @@ def gauge(self, key: str, value: float) -> None:
 
 class DummyStatsLogger(BaseStatsLogger):
     def incr(self, key: str) -> None:
-        logger.debug(Fore.CYAN + "[stats_logger] (incr) " + key + Style.RESET_ALL)
+        if self.should_log(key):
+            logger.debug(Fore.CYAN + "[stats_logger] (incr) " + key + Style.RESET_ALL)
 
     def decr(self, key: str) -> None:
-        logger.debug(Fore.CYAN + "[stats_logger] (decr) " + key + Style.RESET_ALL)
+        if self.should_log(key):
+            logger.debug(Fore.CYAN + "[stats_logger] (decr) " + key + Style.RESET_ALL)
 
     def timing(self, key: str, value: float) -> None:
-        logger.debug(
-            Fore.CYAN + f"[stats_logger] (timing) {key} | {value} " + Style.RESET_ALL
-        )
+        if self.should_log(key):
+            logger.debug(
+                Fore.CYAN
+                + f"[stats_logger] (timing) {key} | {value} "
+                + Style.RESET_ALL
+            )
 
     def gauge(self, key: str, value: float) -> None:
-        logger.debug(
-            Fore.CYAN
-            + "[stats_logger] (gauge) "
-            + f"{key}"
-            + f"{value}"
-            + Style.RESET_ALL
-        )
+        if self.should_log(key):
```

Comment:
@villebro so anyone who wants to use this denylist is going to have to implement the functionality in their own statslogger if they don't use the dummy. Should we put something in UPDATING to explain how to do this or do you think it's evident enough when reading this local example?

Reply:
You're right, maybe we should just add it to StatsdStatsLogger, and remove it from BaseStatsLogger. This will cover the vanilla StatsD use case, and should serve as an example for anyone who wants similar functionality in their own custom logger class. Thoughts @eschutho

Reply:
Yeah, that sounds like a good idea.

Reply:
+1 to comments above.
I don't find much value in having the deny list and find `def should_log` sufficient for filtering and good enough example on how to define a custom logger with filtering.

But that's just a minor preference and current solution is fine by me as well